cabal-install/src/Distribution/Client/Init/Interactive/Command.hs

```diff
@@ -357,12 +357,12 @@ licensePrompt flags = getLicense flags $ do
 authorPrompt :: Interactive m => InitFlags -> m String
 authorPrompt flags = getAuthor flags $ do
     name <- guessAuthorName
-    promptStr "Author name" (DefaultPrompt name)
+    promptOrDefault name "Author name"
```

Comment:
Much better! Do you want to flip the arguments of `promptOrDefault` and replace the whole `do` block with `guessAuthorName >>= promptOrDefault "Author name"`? (Same for email.)